src/components/accounts/EmailSettings.js

```diff
@@ -74,19 +88,49 @@ class EmailSettings extends Component {
     handleEmailChangeSubmit(event) {
         event.preventDefault();
 
-        const isNewEmailValid = this.handleEmailValidation(this.state.newEmail);
-        if (isNewEmailValid) {
-            this.updateEmail()
-                .then(isUpdated => {
-                    this.setState({ failed: !isUpdated });
+        let isCurrentPasswordValid, failure;
+        const isNewEmailValid = this.handleEmailValidation(this.state.newEmail); 
+
+        if(isNewEmailValid){
+            this.validateCurrentPassword(this.state.currentPassword)
```

Comment:
I don't think we need to make a call to check the password before making the call to update email because we're sending the password with the email anyway. If it is invalid, then the `updateEmail` call will fail.